app/components/footer.tsx

```diff
@@ -0,0 +1,202 @@
+        </div>
+      </form>
+    </div>
+  )
+}
+function ContactSection() {
+  return (
+    <div>
+      <H6>Contact</H6>
+      <ul className="mt-4">
+        <FooterLink name="contact page" href="/contact" />
+        <FooterLink name="office hours" href="/office-hours" />
+        <FooterLink name="Join discord" href="https://kcd.im/discord" />
+      </ul>
+    </div>
+  )
+}
+function GeneralSection() {
+  return (
+    <div>
+      <H6>General</H6>
+      <ul className="mt-4">
+        <FooterLink name="privacy policy" href="/privacy" />
+        <FooterLink name="terms of use" href="/terms" />
+        <FooterLink name="code of conduct" href="/coc" />
+      </ul>
+    </div>
+  )
+}
+
+function SitemapSection() {
+  return (
+    <div>
+      <H6>Sitemap</H6>
+      <ul className="mt-4">
+        <FooterLink name="home" href="/legal/terms" />
+        <FooterLink name="blog" href="/legal/privacy" />
+        <FooterLink name="courses" href="/contact" />
+        <FooterLink name="discord" href="/contact" />
+        <FooterLink name="podcast" href="/contact" />
+        <FooterLink name="workshops" href="/contact" />
+        <FooterLink name="about" href="/contact" />
+        <FooterLink name="all pages" href="/contact" />
+      </ul>
+    </div>
+  )
+}
+
+function AboutSection() {
+  return (
+    <div>
+      <H4>Kent C. Dodds</H4>
+
+      <p className="mt-6 max-w-md dark:text-blueGray-500 text-gray-500 text-2xl">
+        Full time educator teaching people development
+      </p>
+
+      <div className="flex mt-6 dark:text-blueGray-500 text-gray-500 space-x-4">
+        <GithubLink />
+        <YoutubeLink />
+        <TwitterLink />
+      </div>
+    </div>
+  )
+}
+
+function FooterLink({name, href}: FooterLinkProps) {
+  return (
+    <li>
+      <a
+        href={href}
+        className="inline-block py-1 dark:text-blueGray-500 text-gray-500 hover:underline whitespace-nowrap text-lg"
+      >
+        {name}
+      </a>
+    </li>
+  )
+}
+
+function GithubLink() {
+  return (
+    <a href="https://kcd.im/github">
+      <svg className="w-8 h-8" viewBox="0 0 24 24">
+        <path
+          fill="currentColor"
+          d="M12,2A10,10 0 0,0 2,12C2,16.42 4.87,20.17 8.84,21.5C9.34,21.58 9.5,21.27 9.5,21C9.5,20.77 9.5,20.14 9.5,19.31C6.73,19.91 6.14,17.97 6.14,17.97C5.68,16.81 5.03,16.5 5.03,16.5C4.12,15.88 5.1,15.9 5.1,15.9C6.1,15.97 6.63,16.93 6.63,16.93C7.5,18.45 8.97,18 9.54,17.76C9.63,17.11 9.89,16.67 10.17,16.42C7.95,16.17 5.62,15.31 5.62,11.5C5.62,10.39 6,9.5 6.65,8.79C6.55,8.54 6.2,7.5 6.75,6.15C6.75,6.15 7.59,5.88 9.5,7.17C10.29,6.95 11.15,6.84 12,6.84C12.85,6.84 13.71,6.95 14.5,7.17C16.41,5.88 17.25,6.15 17.25,6.15C17.8,7.5 17.45,8.54 17.35,8.79C18,9.5 18.38,10.39 18.38,11.5C18.38,15.32 16.04,16.16 13.81,16.41C14.17,16.72 14.5,17.33 14.5,18.26C14.5,19.6 14.5,20.68 14.5,21C14.5,21.27 14.66,21.59 15.17,21.5C19.14,20.16 22,16.42 22,12A10,10 0 0,0 12,2Z"
+        />
+      </svg>
```

Comment:
Note to self, for consistency, this should be extracted to an "icon component" and placed under `/components/icons`. The link can then be inlined in the footer.